kv/migration/migration.go

```diff
@@ -249,12 +249,31 @@ func (m *Migrator) Down(ctx context.Context) (err error) {
 		return wrapErr(err)
 	}
 
-	migrationsToDo := len(migrations)
-	if migrationsToDo > 0 {
-		m.logger.Info("Tearing down metadata migrations", zap.Int("migration_count", migrationsToDo))
+	migrationsToDo := len(migrations) - untilMigration
+	if migrationsToDo == 0 {
```

Comment:
Need to check if `migrationsToDo` is < 0? It would be strange for this to happen, but I think it could happen if somebody is doing a big jump in versions and for some reason runs `influxd downgrade` to an intermediate version. Might be good to log a message of some kind in that case and return early.